terra_layer/models.py

```diff
@@ -12,21 +12,61 @@
 
 
 class Scene(models.Model):
+    """ A scene is a group of data visualisation in terra-visu.
+    It's also a main menu entry.
+    """
+
     name = models.CharField(max_length=255, unique=True)
     slug = models.SlugField(max_length=255, unique=True)
     category = models.CharField(max_length=255, default="map")
     custom_icon = models.ImageField(
         max_length=255, upload_to="scene-icons", null=True, default=None
     )
     order = models.IntegerField(default=0, db_index=True)
+    tree = JSONField(default=list)
 
     def get_absolute_url(self):
         return reverse("scene-detail", args=[self.pk])
 
+    def tree2models(self, current_node=None, parent=None, order=0):
+
+        # Init case, we've just launch the process
+        if current_node is None:
+            current_node = self.tree
+            self.layer_groups.all().delete()  # Clear all groups to generate brand new one
+
+        if not parent:
+            # Create a default unique parent group that is ignored at export
+            parent = LayerGroup.objects.create(view=self, label="Root")
+
+        if isinstance(current_node, list):
+            for idx, child in enumerate(current_node):
+                self.tree2models(current_node=child, parent=parent, order=idx)
+
+        elif "group" in current_node:
+            # Handle groups
+            group = parent.children.create(
+                view=self,
+                label=current_node["title"],
+                exclusive=current_node.get("exclusive", False),
+                order=order,
+            )
+
+            self.tree2models(current_node=current_node["children"], parent=group)
+
+        elif "geolayer" in current_node:
+            # Handle layers
+            layer = Layer.objects.get(pk=current_node["geolayer"])
```

Comment:
- What if layer does not exists (race condition) ?
- What if layer is already referenced in another group (duplicate in the tree) ?

Reply:
Should be forbidden in both cases, I add some cheks.